clients/src/main/java/org/apache/kafka/common/acl/AclOperation.java

```diff
@@ -22,6 +22,24 @@
 
 /**
  * Represents an operation which an ACL grants or denies permission to perform.
+ *
+ * Some operations imply other operations.
+ *
+ * ALLOW ALL implies ALLOW everything
+ * DENY ALL implies DENY everything
+ *
+ * ALLOW READ implies ALLOW DESCRIBE
+ * ALLOW WRITE implies ALLOW DESCRIBE
+ * ALLOW DELETE implies ALLOW DESCRIBE
+ *
+ * ALLOW ALTER implies ALLOW DESCRIBE
+ * DENY DESCRIBE implies DENY ALTER
+ *
+ * ALLOW ALTER_CONFIGS implies ALLOW DESCRIBE_CONFIGS
+ * DENY DESCRIBE_CONFIGS implies DENY ALTER_CONFIGS
+ *
+ * ALLOW ALTER_CONFIGS on kafka-cluster implies ALLOW ALTER_CONFIGS on all topics.
+ * DENY DESCRIBE_CONFIGS on kafka-cluster implies DENY DESCRIBE_CONFIGS on all topics.
```

Comment:
Looks good overall. A couple of comments. (1) My understanding is that describe_configs on cluster just gives the permission to describing the broker level config, not the topic level config. So, describe_configs on cluster seems independent of describe_configs on topic and the former probably shouldn't imply the latter. One can always give describe_configs permission on Topic(*) for access to all topics. Ditto for alter_configs on cluster. (2) Currently, describe is only implied on allow rules. For consistency, it's probably better to keep it that way for now. If there is a need to extend that on deny rules, we can add the extension consistently to all resources.